### PR TITLE
[DO NOT MERGE] Tendermint v0.35.7

### PR DESCRIPTION
**Pull request by Karrenbelt**
# IMPORTANT

[Tendermint v0.35 will be discontinued](https://medium.com/the-interchain-foundation/discontinuing-tendermint-v0-35-a-postmortem-on-the-new-networking-layer-3696c811dabc)

this PR will likely be closed soon:tm:

## Proposed changes

Upgrade from Tendermint v0.34.19 to v0.35.7 (current latest release).

references
- [upgrading](https://github.com/tendermint/tendermint/blob/master/UPGRADING.md#v035)
- [change log](https://github.com/tendermint/tendermint/blob/master/CHANGELOG.md)
- [docker file v0.35.7](https://raw.githubusercontent.com/tendermint/tendermint/v0.35.7/DOCKER/Dockerfile)

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .github/workflows/main_workflow.yml [hunk 1]
```diff
@@ -97,6 +97,7 @@ jobs:
         wget https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-linux-x86_64.zip
         unzip protoc-3.19.4-linux-x86_64.zip -d protoc
         sudo mv protoc/bin/protoc /usr/local/bin/protoc
+        sudo mv protoc/include/* /usr/local/include/
```
Comment (Karrenbelt) [inline .github/workflows/main_workflow.yml:100]: otherwise
```
google/protobuf/descriptor.proto: File not found.
gogoproto/gogo.proto:32:1: Import "google/protobuf/descriptor.proto" was not found or had errors.
...
```

File: docs/api/protocols/abci/custom_types.md [hunk 2]
```diff
@@ -443,7 +443,7 @@ This class represents an instance of EventAttribute.
 #### `__`init`__`
 
 ```python
-def __init__(key: bytes, value: bytes, index: bool)
+def __init__(key: str, value: str, index: bool)
```
Comment (Karrenbelt) [inline docs/api/protocols/abci/custom_types.md:446]: old: https://github.com/valory-xyz/open-autonomy/blob/2822968f858def392ebfd4d48906ce1a81548234/packages/valory/connections/abci/protos/tendermint/abci/types.proto#L315-L320
new: [doesn't display directly](https://github.com/tendermint/tendermint/blob/9daea43375e06816a44c30bdb7b02701df4a5f76/proto/tendermint/abci/types.proto#L282-L286)

File: packages/valory/connections/abci/protos/tendermint/abci/types.proto [hunk 3]
```diff
@@ -163,25 +154,18 @@ message ResponseFlush {}
...
 }
 
-// nondeterministic
-message ResponseSetOption {
-  uint32 code = 1;
-  // bytes data = 2;
-  string log  = 3;
-  string info = 4;
-}
-
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/protos/tendermint/abci/types.proto:180]: this got removed on update to v0.35.7

File: packages/valory/connections/abci/protos/tendermint/abci/types.proto [hunk 4]
```diff
@@ -279,23 +265,6 @@ message ResponseApplySnapshotChunk {
...
-// BlockParams contains limits on the block size.
-message BlockParams {
-  // Note: must be greater than 0
-  int64 max_bytes = 1;
-  // Note: must be greater or equal to -1
-  int64 max_gas = 2;
-}
-
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/protos/tendermint/abci/types.proto:298]: and some amount of reshuffeling

File: packages/valory/connections/abci/scripts/genproto.py [hunk 5]
```diff
@@ -80,14 +86,6 @@ def generate_proto(source: str) -> None:
...
         print("Generating %s..." % output)
 
-        if not os.path.exists(source):
-            sys.stderr.write("Can't find required file: %s\n" % source)
-            sys.exit(-1)
-
-        if protoc is None:
-            sys.stderr.write("protoc is not installed!\n")
-            sys.exit(-1)
-
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/scripts/genproto.py:90]: these were in the wrong place, I move them up

File: packages/valory/connections/abci/tendermint_encoder.py [hunk 6]
```diff
@@ -223,6 +209,10 @@ def response_check_tx(cls, message: AbciMessage) -> Response:
 
         check_tx.codespace = message.codespace
 
+        check_tx.sender = message.tx_sender
+        check_tx.priority = message.priority
+        # mempool_error is set by Tendermint, not the ABCI App
+
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/tendermint_encoder.py:215]: new fields worthy of taking note of.

File: packages/valory/protocols/abci/custom_types.py [hunk 7]
```diff
@@ -1467,6 +1467,7 @@ class PublicKeyType(Enum):
 
         ed25519 = "ed25519"
         secp256k1 = "secp256k1"
+        sr25519 = "sr25519"
```
Comment (Karrenbelt) [inline packages/valory/protocols/abci/custom_types.py:1470]: [polkadot](https://wiki.polkadot.network/docs/learn-cryptography)

File: tests/test_protocols/test_abci/test_abci_spec.py [hunk 8]
```diff
@@ -62,20 +63,28 @@
 
 
 # tests
-def test_local_types_file_matches_github() -> None:
+def test_local_types_file_matches_github(request_attempts: int = 3) -> None:
```
Comment (Karrenbelt) [inline tests/test_protocols/test_abci/test_abci_spec.py:66]: it failed once on me unnecessarily thus retry granted.

File: tests/test_protocols/test_abci/test_abci.py [hunk 9]
```diff
@@ -171,6 +171,7 @@ def build_message(self) -> AbciMessage:
             version="0.1.0",
             block_version=1,
             p2p_version=1,
+            abci_version="0.35.7",
```
Comment (Karrenbelt) [inline tests/test_protocols/test_abci/test_abci.py:174]: not relevant here, but this is actually `0.17.0` I believe

File: deployments/Dockerfiles/localnode/app.py [hunk 10]
```diff
@@ -41,8 +41,7 @@
 DEFAULT_LOG_FILE = "log.log"
 IS_DEV_MODE = os.environ.get("DEV_MODE", "0") == "1"
 CONFIG_OVERRIDE = [
-    ("fast_sync = true", "fast_sync = false"),
-    ("max_num_outbound_peers = 10", "max_num_outbound_peers = 0"),
+    ("max-num-outbound-peers = 10", "max-num-outbound-peers = 0"),
```
Comment (Karrenbelt) [inline deployments/Dockerfiles/localnode/app.py:44]: - The configuration file field [fastsync] has been renamed to [blocksync].
- The top level configuration file field fast-sync has moved under the new [blocksync] field as blocksync.enable
 (https://github.com/tendermint/tendermint/blob/master/UPGRADING.md#v035)

File: .github/workflows/main_workflow.yml (hunk 1 above)
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:100]: why?

File: deployments/Dockerfiles/localnode/app.py (hunk 10 above)
Comment (DavidMinarsch) [reply deployments/Dockerfiles/localnode/app.py:44]: So where is this configuration then here?

File: tests/test_protocols/test_abci/test_abci.py (hunk 9 above)
Comment (DavidMinarsch) [reply tests/test_protocols/test_abci/test_abci.py:174]: I think this is not the version of the protocol, it's the version of the abci app afaik. Please check

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: .github/workflows/main_workflow.yml (hunk 1 above)
Comment (Karrenbelt) [reply .github/workflows/main_workflow.yml:100]: https://github.com/valory-xyz/open-autonomy/pull/1012#discussion_r906406177

File: deployments/Dockerfiles/localnode/app.py (hunk 10 above)
Comment (Karrenbelt) [reply deployments/Dockerfiles/localnode/app.py:44]: [upgrade](https://github.com/tendermint/tendermint/blob/master/UPGRADING.md#config-changes-1) lists the following

- The configuration file field [fastsync] has been renamed to [blocksync].
- The top level configuration file field fast-sync has moved under the new [blocksync] field as blocksync.enable.
- blocksync.version = "v1" and blocksync.version = "v2" (previously fastsync) are no longer supported. Please use v0 instead. During the v0.35 release cycle, v0 was determined to suit the existing needs and the cost of maintaining the v1 and v2 modules was determined to be greater than necessary.

and the config.toml looks as follows:
```
#######################################################
###       Block Sync Configuration Connections       ###
#######################################################
[blocksync]

# If this node is many blocks behind the tip of the chain, BlockSync
# allows them to catchup quickly by downloading blocks in parallel
# and verifying their commits
enable = true

# Block Sync version to use:
#   1) "v0" (default) - the standard Block Sync implementation
#   2) "v2" - DEPRECATED, please use v0
version = "v0"
```

however, we need to find a better way to overwrite because there is also statesync which also uses the same key

```
#######################################################
###         State Sync Configuration Options        ###
#######################################################
[statesync]
# State sync rapidly bootstraps a new node by discovering, fetching, and restoring a state machine
# snapshot from peers instead of fetching and replaying historical blocks. Requires some peers in
# the network to take and serve state machine snapshots. State sync is not attempted if the node
# has any local state (LastBlockHeight > 0). The node will have a truncated block history,
# starting from the height of the snapshot.
enable = false
```

File: deployments/Dockerfiles/localnode/app.py [hunk 11]
```diff
@@ -40,9 +40,12 @@
...
 DEFAULT_LOG_FILE = "log.log"
 IS_DEV_MODE = os.environ.get("DEV_MODE", "0") == "1"
+# https://docs.tendermint.com/v0.34/tendermint-core/configuration.html
+# https://docs.tendermint.com/v0.35/nodes/configuration.html
 CONFIG_OVERRIDE = [
-    ("fast_sync = true", "fast_sync = false"),
-    ("max_num_outbound_peers = 10", "max_num_outbound_peers = 0"),
+    # ("fast-sync = true", "fast-sync = false"),
+    ("and verifying their commits\nenable = true", "and verifying their commits\nenable = false"),
+    ("max-num-outbound-peers = 10", "max-num-outbound-peers = 0"),
```
Comment (Karrenbelt) [inline deployments/Dockerfiles/localnode/app.py:48]: TODO: this needs a better approach.

File: deployments/Dockerfiles/localnode/config-template.toml [hunk 12]
```diff
@@ -1,4 +1,5 @@
-fast_sync = true
+# fast_sync = true
```
Comment (Karrenbelt) [inline deployments/Dockerfiles/localnode/config-template.toml:1]: TODO: delete

File: packages/valory/connections/abci/connection.py [hunk 13]
```diff
@@ -457,12 +458,15 @@ def __init__(self, params: TendermintParams, logger: Optional[Logger] = None):
...
         self.log_file = os.environ.get("LOG_FILE", DEFAULT_TENDERMINT_LOG_FILE)
+        output = subprocess.check_output(["tendermint", "version"])  # nosec
+        self.logger.info(f"Tendermint version: {output.decode(ENCODING).strip()}")
 
     def _build_init_command(self) -> List[str]:
         """Build the 'init' command."""
         cmd = [
             "tendermint",
             "init",
+            "validator",
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/connection.py:469]: we must specify one of the following: `[validator|full|seed]`

some details: [ADR 052: Tendermint Mode](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-052-tendermint-mode.md)

File: tests/test_deployments/test_app.py [hunk 14]
```diff
@@ -193,10 +193,6 @@ class TestTendermintServerApp(BaseTendermintServerTest):
     def test_files_exist(self) -> None:
         """Test that the necessary files are present"""
 
-        def remove_prefix(text: str, prefix: str) -> str:
-            """str.removeprefix only from python3.9 onward"""
-            return text[text.startswith(prefix) and len(prefix) :]
```
Comment (DavidMinarsch) [inline tests/test_deployments/test_app.py:198]: not used?

File: tests/test_deployments/test_app.py [hunk 15]
```diff
@@ -109,7 +109,7 @@ def setup_class(cls) -> None:
         command = [cls.tendermint, "init", "validator", "--home", cls.tm_home]
         process = subprocess.Popen(command, stderr=subprocess.PIPE)  # nosec
         _, stderr = process.communicate()
-        assert not stderr, stderr
+        logging.debug(stderr)
```
Comment (DavidMinarsch) [inline tests/test_deployments/test_app.py:112]: why the change?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Comment by DavidMinarsch** [comment]
Please extend the handler to satisfy the request info changes.

File: tests/test_deployments/test_app.py (hunk 14 above)
Comment (Karrenbelt) [reply tests/test_deployments/test_app.py:198]: exactly 09f223c37743a45a7ee4ab4e7d765b0ae2d3b09e

File: tests/test_deployments/test_app.py (hunk 15 above)
Comment (Karrenbelt) [reply tests/test_deployments/test_app.py:112]: because I was getting errors. while sorting out configuration, will revert

File: .github/workflows/main_workflow.yml [hunk 16]
```diff
@@ -235,7 +238,7 @@ jobs:
           ipfs init
 
           # install Tendermint
-          wget -O tendermint.tar.gz https://github.com/tendermint/tendermint/releases/download/v0.34.19/tendermint_0.34.19_darwin_amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (Karrenbelt) [reply .github/workflows/main_workflow.yml:238]: ubuntu for everyone :+1:

File: packages/valory/connections/abci/connection.py [hunk 17]
```diff
@@ -111,9 +111,7 @@ class _TendermintABCISerializer:
 
     @classmethod
     def encode_varint(cls, number: int) -> bytes:
-        """Encode a number in varint coding."""
-        # Shift to int64
-        number = number << 1
+        """Encode a number (uint64) in varint coding."""
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/connection.py:114]: Messages are written to a byte stream using uin64 length delimiters instead of int64.

File: packages/valory/connections/abci/connection.py [hunk 18]
```diff
@@ -158,7 +156,7 @@ async def decode_varint(
             raise EOFError()
         if not success:
             raise DecodeVarintError("could not decode varint")
-        return result >> 1
+        return result
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/connection.py:159]: and here

**Review by dvilelaf — COMMENTED** [review]
Looks good in general, although I'm a bit out of the loop here. It seems to be some issue with some tests.

**Comment by Karrenbelt** [comment]
observations from tests

### Ubuntu
- `FAILED tests/test_autonomy/test_cli/test_deploy/test_build/test_image.py::TestBuildImage::test_dev`
- `FAILED tests/test_autonomy/test_cli/test_deploy/test_build/test_image.py::TestBuildImage::test_build_prod`

locally running `autonomy deploy build image "valory/oracle_hardhat" --dev`:

```
[Skaffold] E: Unable to locate package python3.10-dev
[Skaffold] E: Couldn't find any package by glob 'python3.10-dev'
[Skaffold] E: Couldn't find any package by regex 'python3.10-dev'
unable to stream build output: The command '/bin/sh -c sudo apt-get install python3.10 python3.10-dev git -y' returned a non-zero code: 100
[Skaffold] 
Image build failed.
```

### macOS
- `ERROR tests/test_deployments/test_app.py::TestTendermintGentleResetServer::test_gentle_reset`
  this appears flaky

### Windows
- `raise RuntimeError('Event loop is closed')` 
      followed by
     `FAILED tests/test_deployments/test_app.py::TestTendermintLogMessages::test_tendermint_logs`

File: tests/test_agents/test_counter.py [hunk 19]
```diff
@@ -56,6 +56,8 @@ def _send_tx_to_node(
         """
         # at least two hex digits
         tx_arg = "0x{:02x}".format(value)
+        log_message = "requesting broadcast_tx_commit"
+        logging.debug(f"{log_message}: {node_address} {tx_arg}")
         result = requests.get(
             node_address + "/broadcast_tx_commit",
             params=dict(tx=tx_arg),
```
Comment (Karrenbelt) [inline tests/test_agents/test_counter.py:63]: This appears to be the issue for Counter

File: tests/test_connections/test_abci/test_abci.py [hunk 20]
```diff
@@ -496,10 +499,16 @@ def test_ensure_connected_raises_connection_error() -> None:
...
+    """Test encode_varint (uint64) method for _TendermintABCISerializer"""
+    assert _TendermintABCISerializer.encode_varint(1 << 31) == b"\x80\x80\x80\x80\x08"
+    assert (
+        _TendermintABCISerializer.encode_varint(1 << 63)
+        == b"\x80\x80\x80\x80\x80\x80\x80\x80\x80\x01"
+    )
+    assert (
+        _TendermintABCISerializer.encode_varint(1 << 127)
+        == b"\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x02"
+    )
```
Comment (Karrenbelt) [inline tests/test_connections/test_abci/test_abci.py:511]: had to change this. There is another test that covers encoding / decoding:

https://github.com/valory-xyz/open-autonomy/blob/0b885133dafeb3d170649b9f6b2dc526e0f57286/tests/test_connections/test_abci/test_abci.py#L514-L524

**Review by Adamantios — COMMENTED** [review]
Progress looks good so far. 

Re the oracle's failing e2e test, I noticed a lot of round timeouts on CI while getting randomness, so I am not sure if this is related to the PR.

File: packages/valory/connections/abci/connection.py [hunk 21]
```diff
@@ -111,7 +111,9 @@ class _TendermintABCISerializer:
 
     @classmethod
     def encode_varint(cls, number: int) -> bytes:
-        """Encode a number (uint64) in varint coding."""
+        """Encode a number in varint coding."""
+        if not 0 <= number < (1 << 64) - 1:
+            raise DecodeVarintError(f"expecting uint64, got: {number}")
```
Comment (Karrenbelt) [inline packages/valory/connections/abci/connection.py:116]: although I did not show this, if a negative value were to be giving the application would keep hanging and eventually, presumably, crash since the buffer would keep growing

File: tests/test_connections/test_abci/test_abci.py [hunk 22]
```diff
@@ -496,13 +499,16 @@ def test_ensure_connected_raises_connection_error() -> None:
...
-    assert _TendermintABCISerializer.encode_varint(10) == b"\x14"
-    assert _TendermintABCISerializer.encode_varint(70) == b"\x8c\x01"
-    assert _TendermintABCISerializer.encode_varint(130) == b"\x84\x02"
+    """Test encode_varint (uint64) method for _TendermintABCISerializer"""
+    hard_zero_encoded = b"\x00"
+    max_uint32_encoded = b"\xff\xff\xff\xff\x0f"
+    max_uint64_encoded = b"\xff\xff\xff\xff\xff\xff\xff\xff\xff\x01"
+    assert _TendermintABCISerializer.encode_varint(0) == hard_zero_encoded
+    assert _TendermintABCISerializer.encode_varint((1 << 32) - 1) == max_uint32_encoded
+    assert _TendermintABCISerializer.encode_varint((1 << 63) - 1) == max_uint64_encoded
```
Comment (Karrenbelt) [inline tests/test_connections/test_abci/test_abci.py:508]: should be `(1 << 64) - 1` - fixing

**Comment by Karrenbelt** [comment]
```
2022-06-29T22:02:15.3760192Z =========================== short test summary info ============================
2022-06-29T22:02:15.3760462Z FAILED tests/test_agents/test_registration.py::TestRegistrationStartUpFourAgentsCatchUp::test_run[4]
2022-06-29T22:02:15.3760690Z FAILED tests/test_agents/test_registration.py::TestRegistrationStartUpFourAgents::test_run[4]
2022-06-29T22:02:15.3760870Z FAILED tests/test_agents/test_oracle_abci.py::TestAgentCatchup::test_run[4]
2022-06-29T22:02:15.3761091Z FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintResetInterrupt::test_run[4]
2022-06-29T22:02:15.3761424Z FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintResetInterruptNoRejoin::test_run[4]
2022-06-29T22:02:15.3761615Z FAILED tests/test_agents/test_counter.py::TestABCICounterSkillMany::test_run
2022-06-29T22:02:15.3761788Z = 6 failed, 12 passed, 6 skipped, 1264 deselected, 1 warning in 4101.68s (1:08:21) =
2022-06-29T22:02:15.3762577Z ERROR: InvocationError for command /home/runner/work/open-autonomy/open-autonomy/.tox/py3.10/bin/pytest -rfE --doctest-modules autonomy tests/ --cov=autonomy --cov-report=html --cov=packages/valory --cov-report=xml --cov-report=term --cov-report=term-missing --cov-config=.coveragerc -m e2e (exited with code 1)
2022-06-29T22:02:15.3762713Z ___________________________________ summary ____________________________________
2022-06-29T22:02:15.3762806Z ERROR:   py3.10: commands failed
2022-06-29T22:02:15.3766616Z ##[error]Process completed with exit code 1.
```

File: tests/test_protocols/test_abci/test_abci_spec.py [hunk 23]
```diff
@@ -62,20 +63,28 @@
...
             status_code, reason = response.status_code, response.reason
             raise requests.HTTPError(f"{log_msg}: {status_code} ({reason})")
         github_data = response.text
         local_data = file.read_text(encoding=ENCODING)
+        if not github_data == local_data:
+            different.append([file, url])
 
-        assert github_data == local_data, f"mismatch:\n{file}\n{url}"
+    if different:
+        logging.error("\n".join([" =/= ".join(map(str, x)) for x in different]))
```
Comment (Karrenbelt) [inline tests/test_protocols/test_abci/test_abci_spec.py:86]: unnecessary list comprehensions

File: tests/test_agents/test_counter.py (hunk 19 above)
Comment (Karrenbelt) [reply tests/test_agents/test_counter.py]: no more was an issue with image building now resolved (cause: `int64` to `uint64`)

File: tests/test_deployments/test_app.py [hunk 24]
```diff
@@ -267,6 +264,7 @@ class TestTendermintLogMessages(BaseTendermintServerTest):
     """Test Tendermint message logging"""
 
     @wait_for_node_to_run
+    @pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
```
[the remark this answers is not in the document]
Comment (Adamantios) [reply tests/test_deployments/test_app.py]: Skipping tests should always be accompanied by issues.

**Comment by Adamantios** [comment]
I studied the failing oracle test. The agent is unable to catch up for some reason. It might be connected to the following error that happens too often:
```
[agent_00003_4agents_run] Incoming envelope: Envelope(to=valory/oracle_abci:0.1.0, sender=valory/http_client:0.1.0, protocol_specification_id=valory/http:1.0.0, message=Message(sender=valory/http_client:0.1.0,to=valory/oracle_abci:0.1.0,body=b'Traceback (most recent call last):\n  File "/tmp/tmp2_xsvxdf/agent_00003_4agents_run/vendor/valory/connections/http_client/connection.py", line 188, in _http_request_task\n    resp = await asyncio.wait_for(\n  File "/opt/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/asyncio/tasks.py", line 445, in wait_for\n    return fut.result()\n  File "/tmp/tmp2_xsvxdf/agent_00003_4agents_run/vendor/valory/connections/http_client/connection.py", line 233, in _perform_http_request\n    async with session.request(\n  File "/home/runner/work/open-autonomy/open-autonomy/.tox/py3.10/lib/python3.10/
```

Also, check this out:
```
[agent_00003_4agents_run] Incoming envelope: Envelope(to=valory/registration_abci:0.1.0, sender=valory/http_client:0.1.0, protocol_specification_id=valory/http:1.0.0, message=Message(sender=valory/http_client:0.1.0,to=valory/registration_abci:0.1.0,body=b'{\n  "jsonrpc": "2.0",\n  "id": -1,\n  "result": {\n    "node_info": {\n      "protocol_version": {\n        "p2p": "8",\n        "block": "11",\n        "app": "0"\n      },\n      "id": "ae1f83196ca1d3e3dccda8ebb8625d89f2a8c3cb",\n      "listen_addr": "tcp://0.0.0.0:26656",\n      "network": "chain-UYpVLU",\n      "version": "0.35.7",\n      "channels": "4020212223303860616263",\n      "moniker": "node0",\n      "other": {\n        "tx_index": "on",\n        "rpc_address": "tcp://0.0.0.0:26657"\n      }\n    },\n    "sync_info": {\n      "latest_block_hash": "ADB898736FC0545CBEBC1254E82ED2,dialogue_reference=('25735d127d1d19cc51a9d61feb79504218802faa63f095edfad1d1b4a6656e7b', '6b52360db8820bed4cd0df440f6c7591c88c0eacc4760d8d96c2d5f3f580d327'),headers=Content-Type: application/json
```
This comes from the `/status` response from Tendermint. How does the status of agent3 return that?
Does this mean that the agent is connected to another node than its own?

**Comment by Karrenbelt** [comment]
In response to the [comment above](https://github.com/valory-xyz/open-autonomy/pull/1012#issuecomment-1172028653) 
I have some more observations to share from running the simple ABCI tests that in #1043 

```
2022-07-01T12:44:25Z INFO starting service impl=MConn{172.17.0.1:48472} module=p2p 
peer={"Hostname":"172.17.0.1","NodeID":"5d23d861ca4730bc614d65a48bec4bc4973ccd8a","Path":"","Port":48472,"Protocol":"mconn"} service=MConnection
2022-07-01T12:44:25Z ERROR failed to accept connection err="peer \"5d23d861ca4730bc614d65a48bec4bc4973ccd8a\" 
is already connected" module=p2p op=incoming/accepted peer=5d23d861ca4730bc614d65a48bec4bc4973ccd8a
```

```
2022-07-01T12:43:37Z ERROR failed to handshake with peer err="expected to connect with peer 
\"4af0c44186488c5e2416e576b891863e232380ed\", got \"f14134fc73c836ac2f64f5e60b470eef2780152b\"" module=p2p 
peer={"Hostname":"node2","NodeID":"4af0c44186488c5e2416e576b891863e232380ed","Path":"","Port":26656,"Protocol":"mconn"}
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0xe764fb]

File: packages/valory/connections/abci/connection.py [hunk 25]
```diff
@@ -112,8 +112,8 @@ class _TendermintABCISerializer:
     @classmethod
     def encode_varint(cls, number: int) -> bytes:
         """Encode a number in varint coding."""
-        # Shift to int64
-        number = number << 1
+        if not 0 <= number < 1 << 64:
+            raise DecodeVarintError(f"expecting uint64, got: {number}")
```
Comment (DavidMinarsch) [inline packages/valory/connections/abci/connection.py:116]:
```suggestion
            raise EncodeVarintError(f"expecting uint64, got: {number}")
```

**Comment by marcofavorito** [comment]
We use `--p2p.persistent-peers` instead of `--p2p.seeds` because with Tendermint v0.35.7 it seems there is a different behaviour of the Tendermint network setup.

This test requires `tendermint` 0.35.7 installed. With version 0.34.19, both configurations seem to work; but with `0.35.7`, only the one with `persistent-peers` works. 

Remember to replace `--log-level` with `--log_level` and `--proxy-app` with `--proxy_app` if you want to test version 0.34.19.

```
############################
# instructions to reproduce issue

# create new directory
mkdir test-tendermint-issue

# initialize testnet in current directory with 2 validators
tendermint testnet --v 2 --o .

# check Tendermint version is 0.35.7
echo "Expected Tendermint version: 0.35.7"
echo "Actual Tendermint version: $(tendermint version)"

#########
# with --p2p.seeds, it does not work

# reset testnet
/bin/rm -r node0 node1
tendermint testnet --v 2 --o .

# terminal 1
export NODE_ID_0=$(tendermint show-node-id --home ./node0 | tail -n 1)
export NODE_ID_1=$(tendermint show-node-id --home ./node1 | tail -n 1)
tendermint start --log-level debug --rpc.laddr tcp://127.0.0.1:26657 --p2p.laddr tcp://0.0.0.0:26656 --home ./node0 --proxy-app=kvstore --p2p.seeds="${NODE_ID_1}@localhost:26666" | tee tendermint-node-00.log

# terminal 2
export NODE_ID_0=$(tendermint show-node-id --home ./node0 | tail -n 1)
export NODE_ID_1=$(tendermint show-node-id --home ./node1 | tail -n 1)
tendermint start --log-level debug --rpc.laddr tcp://127.0.0.1:26667 --p2p.laddr tcp://0.0.0.0:26666 --home ./node1 --proxy-app=kvstore --p2p.seeds="${NODE_ID_0}@localhost:26656" | tee tendermint-node-01.log

# terminal 3
curl -s 'localhost:26657/broadcast_tx_commit?tx="abcd"'  # it does not return

# Tendermint processes keep logging messages of type: no available peers to send a PEX request to (retrying) module=main


#########
# with --p2p.persistent-peers, it works

# reset testnet
/bin/rm -r node0 node1
tendermint testnet --v 2 --o .

# terminal 1
export NODE_ID_0=$(tendermint show-node-id --home ./node0 | tail -n 1)
export NODE_ID_1=$(tendermint show-node-id --home ./node1 | tail -n 1)
tendermint start --log-level debug --rpc.laddr tcp://127.0.0.1:26657 --p2p.laddr tcp://0.0.0.0:26656 --home ./node0 --proxy-app=kvstore --p2p.persistent-peers="${NODE_ID_1}@localhost:26666" | tee tendermint-node-00.log

# terminal 2
export NODE_ID_0=$(tendermint show-node-id --home ./node0 | tail -n 1)
export NODE_ID_1=$(tendermint show-node-id --home ./node1 | tail -n 1)
tendermint start --log-level debug --rpc.laddr tcp://127.0.0.1:26667 --p2p.laddr tcp://0.0.0.0:26666 --home ./node1 --proxy-app=kvstore --p2p.persistent-peers="${NODE_ID_0}@localhost:26656" | tee tendermint-node-01.log

# terminal 3
curl -s 'localhost:26657/broadcast_tx_commit?tx="abcd"'  # it does not return

# OK
```

**Comment by Karrenbelt** [comment]
```
BootstrapPeers has been added as part of the new p2p stack. This will eventually replace Seeds. Bootstrap peers are connected with on startup if needed for peer discovery. Unlike persistent peers, there's no gaurantee that the node will remain connected with these peers.

**Comment by DavidMinarsch** [comment]
Closing in favour of #1255 and the fact that this is discontinued!
